python/cucim/src/cucim/skimage/transform/_geometric.py

```diff
@@ -1102,12 +1102,12 @@ def __call__(self, coords):
         """
 
         xp = cp.get_array_module(coords)
-        if xp == cp:
-            coords = coords.get()
-        out = np.empty_like(coords, np.double)
+        out = xp.empty_like(coords, xp.double)
 
         # determine triangle index for each coordinate
-        simplex = self._tesselation.find_simplex(coords)
+        # coords must be on host for calls to _tesselation methods
+        simplex = self._tesselation.find_simplex(cp.asnumpy(coords))
```

Comment:
Why do we coerce to a NumPy array here?

Note: This comes up again below, but only commenting here for simplicity.